api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -190,7 +189,11 @@ public static boolean isPromotionAllowed(Type from, Type.PrimitiveType to) {
   }
 
   public static class SchemaVisitor<T> {
-    private final Deque<Integer> fieldIds = Lists.newLinkedList();
+    public void beforeField(Types.NestedField field) {
```

Comment:
nit: would be better to name the two as preVisitField(...)  & postVisitField(...) ?

Reply:
Why do you prefer those names? I think they capture the same information.

I'm going to go ahead and merge this since @chenjunjiedada is hitting the problem, but we can rename these later if we think that there is value.